packages/core/src/lib/markdown-it/plugins/markdown-it-alt-frontmatter.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+// Process "---" => "<frontmatter>
+
+module.exports = function centertext_plugin(md) {
+  function alt_frontmatter(state, startLine, endLine, silent) {
+    var marker, len, nextLine, token,
+      haveEndMarker = false,
+      pos = state.bMarks[startLine] + state.tShift[startLine],
+      max = state.eMarks[startLine];
+    const keyValueRegex = new RegExp("^\\w+:\\s.*");
+
+    if (pos + 3 > max) { return false; }
+    marker = state.src.slice(pos, pos + 3);
+    if (marker !== '---') { return false; }
+
+    // Since we found the marker, we can now search for the end marker
+    // starting from the next line
+    nextLine = startLine;
+    for (;;) {
+      nextLine++;
+      if (nextLine >= endLine) {
+        // unclosed block should be autoclosed by end of document.
+        // also block seems to be autoclosed by end of parent
+        break;
+      }
+
+      pos = state.bMarks[nextLine] + state.tShift[nextLine];
+      max = state.eMarks[nextLine];
+
+      if (pos < max && state.sCount[nextLine] < state.blkIndent) {
+        // non-empty line with negative indent should stop the list:
+        // - ```
+        //  test
+        break;
+      }
+
+      if (state.src.slice(pos, max).trim() === '---') {
```

Comment:
Nit: Let's extract `state.src.slice(pos, max).trim()` into a `const` variable since it's used multiple times; same for the string `'---'`

Reply:
Another nit: can `pos` and `max` be renamed more descriptively, like lineStart and lineEnd?